### PR TITLE
Email notification fixes

### PR DESCRIPTION
**Pull request by juyrjola**

### CODE REVIEW

File: resources/templates/mail/reservation_cancelled.jinja [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% trans resource=reservation.resource.name %}Your reservation has been cancelled.{% endtrans %}
```
Comment (tuomas777) [inline resources/templates/mail/reservation_cancelled.jinja]: needless bind `resource`

File: resources/templates/mail/reservation_confirmed.jinja [hunk 2]
```diff
@@ -1,3 +1,10 @@
-{% trans resource=reservation.resource.name %}Your preliminary reservation has been confirmed.{% endtrans %}
+{% trans resource=reservation.resource.name %}Your reservation has been confirmed.{% endtrans %}
```
Comment (tuomas777) [inline resources/templates/mail/reservation_confirmed.jinja]: needless bind `resource`

File: resources/templates/mail/reservation_denied.jinja [hunk 3]
```diff
@@ -1,3 +1,7 @@
-{% trans resource=reservation.resource.name %}Your preliminary reservation to has been denied.{% endtrans %}
+{% trans resource=reservation.resource.name %}Your reservation request has been denied.{% endtrans %}
```
Comment (tuomas777) [inline resources/templates/mail/reservation_denied.jinja]: needless bind `resource`

File: resources/tests/test_reservation_api.py [hunk 4]
```diff
@@ -898,60 +897,70 @@ def test_state_filters(user_api_client, user, list_url, reservations_in_all_stat
 
 @override_settings(RESPA_MAILS_ENABLED=True)
 @pytest.mark.django_db
-def test_requested_mail(staff_api_client, staff_user, list_url, resource_in_unit, reservation,
+def test_requested_mail(staff_api_client, staff_user, user_api_client, list_url,
```
Comment (tuomas777) [inline resources/tests/test_reservation_api.py]: `staff_api_client` isn't used anymore

File: resources/tests/test_reservation_api.py [hunk 5]
```diff
@@ -898,60 +897,70 @@ def test_state_filters(user_api_client, user, list_url, reservations_in_all_stat
...
                         reservation_data_extra):
-    resource_in_unit.need_manual_confirmation = True
-    resource_in_unit.save()
-    assign_perm('can_approve_reservation', staff_user, reservation.resource.unit)
+    resource = Resource.objects.get(id=reservation_data_extra['resource'])
+    resource.need_manual_confirmation = True
+    resource.save()
+    assign_perm('can_approve_reservation', staff_user, resource.unit)
```
Comment (tuomas777) [inline resources/tests/test_reservation_api.py]: hmm, I don't think this or `staff_user`is needed either here anymore

File: resources/tests/test_reservation_api.py [hunk 6]
```diff
@@ -898,60 +897,70 @@ def test_state_filters(user_api_client, user, list_url, reservations_in_all_stat
...
     assert response.status_code == 201
     assert len(mail.outbox) == 1
     mail_instance = mail.outbox[0]
     assert 'Reservation requested' in mail_instance.subject
     assert len(mail_instance.to) == 1
-    assert mail_instance.to[0] == 'test.reserver@test.com'
+    assert mail_instance.to[0] == reservation_data_extra['reserver_email_address']
     mail_message = str(mail_instance.message())
-    assert 'You have created a preliminary reservation' in mail_message
+    assert 'made a preliminary reservation' in mail_message
```
Comment (tuomas777) [inline resources/tests/test_reservation_api.py]: I think these mail related asserts could be done using the helper `check_received_mail()`

**Comment by tuomas777** [comment]
apart from commented minor issues, LGTM

**Comment by juyrjola** [comment]
Merging after fixing review issues.

**Pull request merged**